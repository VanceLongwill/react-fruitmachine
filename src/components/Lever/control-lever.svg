<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 512 512" style="enable-background:new 0 0 512 512;" xml:space="preserve">
<path style="fill:#999999;" d="M428.463,471.59v21.559c0,5.953-4.826,10.78-10.779,10.78h-5.956H100.272h-5.956
	c-5.953,0-10.779-4.826-10.779-10.78V471.59c0-5.953,4.826-10.779,10.779-10.779h15.033l15.485-73.56
	c4.201-19.953,21.803-34.235,42.191-34.235h177.949c20.388,0,37.99,14.282,42.191,34.235l15.485,73.56h15.034
	C423.638,460.811,428.463,465.637,428.463,471.59z"/>
<path style="fill:#6D6D6D;" d="M428.463,471.59v21.559c0,5.953-4.826,10.78-10.779,10.78h-5.956h-26.38
	c5.953,0,10.779-4.826,10.779-10.78V471.59c0-5.953-4.826-10.779-10.779-10.779h-15.033H109.349l6.808-32.339h220.78
	c13.693,0,23.916-12.601,21.095-26l-3.205-15.221c-4.201-19.953-21.803-34.235-42.191-34.235h32.337
	c20.388,0,37.99,14.282,42.191,34.235l15.485,73.56h15.034C423.638,460.811,428.463,465.637,428.463,471.59z"/>
<path id="lever" style="fill:#BFBFBF;" d="M275.409,358.2l-41.647,11.159l-69.745-260.305l41.647-11.159L275.409,358.2z"/>
<path id="lever" style="fill:#6D6D6D;" d="M207.251,103.822L275.408,358.2l-19.527,5.232l-68.156-254.378L207.251,103.822z"/>
<path id="lever" style="fill:#FFB933;" d="M222.558,115.688l-64.277,17.224c-5.385,1.443-10.982-1.458-12.906-6.69L126.279,74.3
	c-9.991-27.164,5.498-57.043,33.454-64.534l0,0c27.955-7.491,56.308,10.64,61.237,39.16l9.421,54.513
	C231.34,108.933,227.944,114.245,222.558,115.688z"/>
<path id="lever" style="fill:#FFD34E;" d="M193.113,123.578l-34.831,9.333c-5.385,1.443-10.982-1.458-12.906-6.69L126.279,74.3
	c-8.526-23.181,1.513-48.327,22.011-59.862c17.115,8.417,30.212,24.603,33.745,45.041L193.113,123.578z"/>
<path id="lever" style="fill:#BFBFBF;" d="M334.068,348.24c11.556,43.129-14.037,87.459-57.164,99.015s-87.455-14.038-99.011-57.167
	c-11.556-43.129,14.037-87.459,57.164-99.015C278.183,279.517,322.512,305.111,334.068,348.24z"/>
<path id="lever" style="fill:#999999;" d="M292.421,359.401c5.393,20.126-6.55,40.814-26.677,46.207c-20.125,5.393-40.812-6.551-46.205-26.678
	s6.55-40.814,26.677-46.207C266.341,327.329,287.028,339.273,292.421,359.401z"/>
<path style="fill:#444444;" d="M265.765,405.608c-7.983,2.14-16.051,1.542-23.221-1.203c17.85-6.805,28.011-26.15,22.96-45.004
	c-3.253-12.144-12.078-21.299-22.984-25.475c1.205-0.459,2.444-0.862,3.717-1.203c20.125-5.393,40.812,6.551,46.205,26.678
	C297.833,379.527,285.89,400.215,265.765,405.608z"/>
<!--
<path d="M417.684,452.713h-8.474l-14.134-67.14c-4.959-23.557-26.029-40.655-50.102-40.655h-3.433
	c-10.373-36.651-42.501-61.557-78.582-64.429l-41.815-156.066l3.506-0.94c9.485-2.542,15.378-11.757,13.705-21.434l-9.421-54.513
	C226.2,31.709,216.871,17.669,203.34,9.017c-13.53-8.651-30.189-11.23-45.699-7.073c-15.513,4.157-28.65,14.719-36.042,28.978
	c-7.391,14.259-8.451,31.082-2.907,46.157L137.789,129c3.391,9.218,13.101,14.251,22.585,11.709l3.506-0.94l41.807,156.035
	c-17.202,11.818-29.607,29.327-35.248,49.115h-3.414c-24.073,0-45.142,17.098-50.102,40.655l-14.134,67.14h-8.473
	c-10.401,0-18.863,8.463-18.863,18.864v21.559c0,10.401,8.463,18.863,18.863,18.863h237.137c4.465,0,8.084-3.62,8.084-8.084
	s-3.62-8.084-8.084-8.084H94.316c-1.485,0-2.695-1.209-2.695-2.695v-21.559c0-1.485,1.209-2.695,2.695-2.695h75.453
	c4.465,0,8.084-3.62,8.084-8.084c0-4.465-3.62-8.084-8.084-8.084h-50.456l13.434-63.809c3.392-16.118,17.809-27.816,34.279-27.816
	h0.389c-0.937,10.166-0.124,20.659,2.669,31.081c6.148,22.945,20.863,42.123,41.433,54c20.571,11.877,44.535,15.032,67.478,8.884
	c22.944-6.148,42.121-20.864,53.997-41.435c9.314-16.132,13.263-34.352,11.57-52.53h0.412c16.47,0,30.887,11.698,34.279,27.816
	l13.434,63.809h-50.455c-4.465,0-8.084,3.62-8.084,8.084c0,4.465,3.62,8.084,8.084,8.084h75.453c1.485,0,2.695,1.209,2.695,2.695
	v21.559c0,1.485-1.209,2.695-2.695,2.695H363.79c-4.465,0-8.084,3.62-8.084,8.084s3.62,8.084,8.084,8.084h53.895
	c10.401,0,18.863-8.463,18.863-18.864v-21.559C436.548,461.175,428.085,452.713,417.684,452.713z M167.504,122.057
	c-0.02,0.005-0.04,0.011-0.06,0.016l-11.255,3.016c-1.355,0.361-2.742-0.356-3.226-1.673l-19.097-51.92
	c-4.039-10.981-3.297-22.747,2.087-33.132c5.384-10.387,14.572-17.773,25.872-20.801c3.58-0.959,7.195-1.434,10.778-1.434
	c7.724,0,15.294,2.205,22.028,6.51c9.856,6.302,16.381,16.122,18.373,27.65l9.421,54.513c0.239,1.383-0.603,2.699-1.957,3.061
	l-52.902,14.176C167.544,122.046,167.523,122.051,167.504,122.057z M179.498,135.583l26.029-6.975l40.762,152.135
	c-4.432,0.489-8.887,1.319-13.325,2.508c-4.409,1.181-8.649,2.675-12.708,4.453L179.498,135.583z M318.99,405.532
	c-9.717,16.831-25.407,28.871-44.18,33.901c-18.772,5.031-38.377,2.449-55.21-7.268c-16.83-9.717-28.869-25.408-33.9-44.182
	c-10.383-38.754,12.695-78.729,51.447-89.114c38.752-10.387,78.726,12.697,89.11,51.449
	C331.289,369.093,328.707,388.701,318.99,405.532z M288.288,336.673c8.707,8.662,13.503,20.197,13.503,32.479
	c0,25.262-20.55,45.813-45.811,45.813c-4.465,0-8.084-3.62-8.084-8.084c0-4.465,3.62-8.084,8.084-8.084
	c16.345,0,29.642-13.298,29.642-29.643c0-7.947-3.103-15.411-8.737-21.017c-3.166-3.148-3.179-8.267-0.029-11.433
	C280.004,333.537,285.123,333.524,288.288,336.673z M235.02,390.113c3.157,3.157,3.157,8.276,0,11.433
	c-1.578,1.578-3.648,2.368-5.716,2.368c-2.068,0-4.138-0.789-5.716-2.368c-8.652-8.652-13.418-20.158-13.418-32.394
	c0-25.262,20.55-45.813,45.811-45.813c4.465,0,8.084,3.62,8.084,8.084c0,4.465-3.62,8.084-8.084,8.084
	c-16.345,0-29.642,13.298-29.642,29.643C226.338,377.071,229.421,384.514,235.02,390.113z"/>
-->
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
</svg>
