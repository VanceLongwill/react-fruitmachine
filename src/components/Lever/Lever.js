import React from 'react'
import styled, { keyframes } from 'styled-components'

import { ReactComponent as LeverSVG } from './control-lever.svg'

const leverAnimation = keyframes`
  0% {
    transform: rotateZ(0);
  }
  50% {
    transform: rotateZ(30deg);
  }
  100% {
    transform: rotateZ(0);
  }
`

const AnimatedLever = styled.div`
  transform: rotate(90deg);
  position: absolute;
  margin-left: 400px;
  margin-top: -160px;
  height: 200px;
  width: 200px;
  #lever { /* the elements we want to animate are tagged with id="lever" in LeverSVG */
    transform-origin: 50% 72%;
    animation-iteration-count: 1; 
    animation-timing-function: ease;
    animation-duration: 1s;
    animation-fill-mode: forwards;
    animation-name: ${p => p.isSpinning ? leverAnimation : 'none'};
  }
`

export function Lever(props) {
  const { isSpinning, onClick } = props
  return <AnimatedLever onClick={onClick} isSpinning={isSpinning}><LeverSVG /></AnimatedLever>
}
