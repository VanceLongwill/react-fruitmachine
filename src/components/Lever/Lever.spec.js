import React from 'react'
import { mount } from 'enzyme'

import { Lever } from './Lever'

describe('Lever', () => {
  it('should match snapshot', () => {
    expect(mount(<Lever isSpinning={false} onClick={jest.fn()} />)).toMatchSnapshot()
  })
})
