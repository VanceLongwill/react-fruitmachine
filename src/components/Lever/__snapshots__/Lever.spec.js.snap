// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lever should match snapshot 1`] = `
<styled.div
  isSpinning={false}
  onClick={[MockFunction]}
>
  <div
    className="sc-bdVaJa ksWKUd"
    onClick={[MockFunction]}
  >
    <svg>
      control-lever.svg
    </svg>
  </div>
</styled.div>
`;
